volatility/framework/plugins/mac/list_files.py

```diff
@@ -0,0 +1,187 @@
+            if v_name == None:
+                return added
+
+            parent = cls._get_parent(vnode)
+            if parent:
+                parent_val = parent.dereference().vol.offset
+            else:
+                parent_val = None
+        
+            loop_vnodes[key] = (v_name, parent_val, vnode)
+
+            added = True
+
+        return added
+
+    @classmethod
+    def _walk_vnode(cls, vnode, loop_vnodes):
+        while vnode:
+            if not cls._add_vnode(vnode, loop_vnodes):
+                break
+
+            parent = cls._get_parent(vnode)
+            while parent:
+                cls._walk_vnode(parent, loop_vnodes)
+                parent = cls._get_parent(parent)
+
+            try:
+                vnode = vnode.v_mntvnodes.tqe_next
+            except exceptions.InvalidAddressException:
+                break
+
+    @classmethod
+    def _process_vnode(cls, vnode, loop_vnodes):
+        cls._walk_vnode(vnode, loop_vnodes)
+   
+    @classmethod
+    def _walk_vnodelist(cls, list_head, loop_vnodes):
+        for vnode in mac.MacUtilities.walk_tailq(list_head,  "v_mntvnodes"):
+            cls._process_vnode(vnode, loop_vnodes)
+
+    @classmethod
+    def _walk_mounts(cls,
+                   context: interfaces.context.ContextInterface,
+                   layer_name: str,
+                   darwin_symbols: str) -> \
+            Iterable[interfaces.objects.ObjectInterface]:
+        
+        loop_vnodes = {}
+
+        # iterate each vnode source from each mount
+        list_mounts = mount.Mount.list_mounts(context, layer_name, darwin_symbols)
+        for mnt in list_mounts:
+            cls._walk_vnodelist(mnt.mnt_vnodelist,   loop_vnodes)
+            cls._walk_vnodelist(mnt.mnt_workerqueue, loop_vnodes)
+            cls._walk_vnodelist(mnt.mnt_newvnodes,   loop_vnodes)
+
+            cls._process_vnode(mnt.mnt_vnodecovered, loop_vnodes)
+            cls._process_vnode(mnt.mnt_realrootvp, loop_vnodes)
+            cls._process_vnode(mnt.mnt_devvp, loop_vnodes)
+           
+        return loop_vnodes
+
+    @classmethod
+    def _build_path(cls, vnodes, vnode_name, parent_offset):
+        path = [vnode_name]
+        
+        while parent_offset in vnodes:
+            parent_name, parent_offset, _ = vnodes[parent_offset]
+            if parent_offset == None:
+                parent_offset = 0
+            
+            path.insert(0, parent_name)
+
+        if len(path) > 1:
+            path = "/".join(path)
+        else:
+            path = vnode_name
+
+        return path
+    
+    @classmethod
+    def list_files(cls,
+                   context: interfaces.context.ContextInterface,
+                   layer_name: str,
+                   darwin_symbols: str) -> \
+            Iterable[interfaces.objects.ObjectInterface]:
+
+        vnodes = cls._walk_mounts(context, layer_name, darwin_symbols)
+
+        dprint("got %d vnodes" % len(vnodes))
```

Comment:
Given the function shouldn't really stay in here, we either need to bulk the function up if a quick debugging mechanism is needed, or remove the calls to it too...

Reply:
Done.